pkg/network/stats.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"time"
+)
+
+// ReqEvent represents either an incoming or closed request.
+// +k8s:deepcopy-gen=false
+type ReqEvent struct {
+	Time time.Time
+	Type ReqEventType
+}
+
+// ReqEventType denotes the type (incoming/closed) of a ReqEvent.
+type ReqEventType int
+
+const (
+	// ReqIn represents an incoming request
+	ReqIn ReqEventType = iota
+	// ReqOut represents a finished request
+	ReqOut
+	// ProxiedIn represents an incoming request through a proxy.
+	ProxiedIn
+	// ProxiedOut represents a finished proxied request.
+	ProxiedOut
+)
+
+// NewRequestStats builds a RequestStats instance, started at the given time.
+func NewRequestStats(startedAt time.Time) *RequestStats {
```

Comment:
Why pointer?

Reply:
Because stuff is mutable.

Reply:
Yeah, but it's not stored in a map or slice, it's just a state.
In fact it's on a stack in a go func (`state := network.NewRequestStats(startedAt)`)
So it won't escape the stack?

Reply:
🤦